src/lib/operators/jit_operator/operators/jit_aggregate.cpp

```diff
@@ -0,0 +1,312 @@
+      if (!jit_aggregate_equals(_groupby_columns[i].tuple_value, _groupby_columns[i].hashmap_value, index, context)) {
+        all_values_equal = false;
+        break;
+      }
+    }
+    // If all values match, we have identified the group the current tuple belongs to.
+    if (all_values_equal) {
+      found_match = true;
+      row_index = index;
+      break;
+    }
+  }
+
+  // If no row matches, a new tuple group must be created.
+  // This requires adding a row to each output vector.
+  // For groupby columns, this new row contains the value from the current tuple.
+  // Aggregates are initialized with a value that is specific to their aggregate function.
+  // The it_grow_by_one function appends an element to the end of an output vector and returns the index of that newly
+  // added value in the vector.
+  if (!found_match) {
+    for (uint32_t i = 0; i < num_groupby_columns; ++i) {
+      // Grow each groupby column vector and copy the value from the current tuple.
+      row_index = jit_grow_by_one(_groupby_columns[i].hashmap_value, JitVariantVector::InitialValue::Zero, context);
+      jit_assign(_groupby_columns[i].tuple_value, _groupby_columns[i].hashmap_value, row_index, context);
+    }
+    for (uint32_t i = 0; i < num_aggregate_columns; ++i) {
+      // Grow each aggregate column vector and initialize the aggregate with a proper initial value.
+      switch (_aggregate_columns[i].function) {
+        case AggregateFunction::Count:
+        case AggregateFunction::Sum:
+          row_index =
+              jit_grow_by_one(_aggregate_columns[i].hashmap_value, JitVariantVector::InitialValue::Zero, context);
+          break;
+        case AggregateFunction::Max:
+          row_index =
+              jit_grow_by_one(_aggregate_columns[i].hashmap_value, JitVariantVector::InitialValue::MinValue, context);
+          break;
+        case AggregateFunction::Min:
+          row_index =
+              jit_grow_by_one(_aggregate_columns[i].hashmap_value, JitVariantVector::InitialValue::MaxValue, context);
+          break;
+        case AggregateFunction::Avg:
+          row_index =
+              jit_grow_by_one(_aggregate_columns[i].hashmap_value, JitVariantVector::InitialValue::Zero, context);
+          DebugAssert(_aggregate_columns[i].hashmap_value_2, "Invalid avg aggregate column.");
+          jit_grow_by_one(_aggregate_columns[i].hashmap_value_2.value(), JitVariantVector::InitialValue::Zero, context);
+          break;
+        case AggregateFunction::CountDistinct:
+          Fail("Not supported");
+      }
+    }
+
+    // Add the index of the new tuple group to the hashmap.
+    hash_bucket.push_back(row_index);
+  }
+
+  // Step 3: Update the aggregate values by calling jit_aggregate_compute with appropriate operation lambdas.
+  for (uint32_t i = 0; i < num_aggregate_columns; ++i) {
+    switch (_aggregate_columns[i].function) {
+      case AggregateFunction::Count:
```

Comment:
Do you correctly handle the case of empty input tables where count should return 0? See #869 and #884.

Reply:
Currently this is not handled. What would be the expected result for `SELECT MAX(a) FROM X GROUP BY b` for an empty table? `SELECT b, MAX(a) FROM X GROUP BY b`?

Reply:
The JIT code now behaves like `mrks/somefixes` (#884) on empty tables (i.e., one output row for no groupby columns and no outputs if there are groupby columns). Also added tests for this.